builtin/common/vector.lua

```diff
@@ -131,3 +131,74 @@ function vector.divide(a, b)
 	end
 end
 
+-- returns a new normalized 2d vector with random values
+function vector.random2d()
+	local vect
+	repeat
+		vect = {x=math.random()*2-1, y=0, z=math.random()*2-1}
+	until vector.length(vect) <= 1
+	return vector.normalize(vect)
+end
+
+-- returns a new normalized 3d vector with random values
+function vector.random3d()
+	local vect
+	repeat
+		vect = {x=math.random()*2-1, y=math.random()*2-1, z=math.random()*2-1}
+	until vector.length(vect) <= 1
+	return vector.normalize(vect)
+end
+
+-- returns a vector with a specific length
+function vector.set_length(v, l)
+	v = vector.normalize(v)
+	return vector.multiply(v, l)
```

Comment:
set_length means you want to give the vector a specific length, but you can't set the length of the zero vector to something not zero